Network/Network.cpp

```diff
@@ -213,12 +222,14 @@ namespace WPEFramework
                     IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_INTERFACE_IPADDRESS, eventHandler) );
                     IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_DEFAULT_INTERFACE, eventHandler) );
                     LOGINFO("Successfully activated Network Plugin");
+                    isNetworkPluginInitialized = true;
                 }
             }
             else
             {
                 msg = "IARM bus is not available";
                 LOGERR("IARM bus is not available. Failed to activate Network Plugin");
+                retryIarmEventRegistration();
```

Comment:
This is WRONG PLACE... 
It should be called when (retVal != IARM_RESULT_SUCCESS) case. 

ie Line#217